hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/SparkBucketIndexPartitioner.java

```diff
@@ -130,6 +131,9 @@ public BucketInfo getBucketInfo(int bucketNumber) {
       // Always write into log file instead of base file if using NB-CC
       BucketType bucketType = isNonBlockingConcurrencyControl ? BucketType.UPDATE : BucketType.INSERT;
       String fileIdPrefix = BucketIdentifier.newBucketFileIdPrefix(bucketId, isNonBlockingConcurrencyControl);
+      if (isNonBlockingConcurrencyControl) {
+        fileIdPrefix = FSUtils.createNewFileId(fileIdPrefix, 0);
+      }
```

Comment:
Could this be combined with L133 since L133 also uses the flag `isNonBlockingConcurrencyControl`?

Reply:
@danny0405 I have another question.  Does this mean that the file ID change between single writer and NBCC?  Would that cause issues on the same table if the user switches the concurrency control mechanism?

Reply:
> Could this be combined with L133 since L133 also uses the flag `isNonBlockingConcurrencyControl`?

updated

Reply:
Why we need this file number then, the file number is introduced to avoid file name conflicts for OCC, but since 1.x, the file name contains current instant time so there should be no chance to conflict.

Reply:
@yihua The NBCC writer would skip the conflict resolution, but the OCC writer would still resolve conflicts with OCC strategies. For single insert writer, there should be no conflicts if there is not small file group merging.

But generally the NB-CC is just designed for upsert scenarios, the combination of NBCC and OCC is not suggested.